doc/developers-guide/accumulators.md

```diff
@@ -30,132 +30,111 @@ a) performance - if we need to do something involving inspecting a stanza or mor
+* Avoid passing superfluous arguments to handlers - e.g. an `LServer` in hook args is redundant since it is already present in the accumulator.
+* Do not use `run` - it is still present in API but executes `run_fold` with `ok` as an initial accumulator anyway.
  Handlers have been rewritten so that they accept an acc as the first arg.
- If you use `run_fold` you have to provide an accumulator, if you use `run` an empty accumulator is created and passed to the handlers.
  Note that `run` is deprecated now and at some point will be removed.
 
 Most handlers have already been modified so that they accept an instance of `mongoose_acc:t()` as the first argument and return value by storing it inside it.
 How the accumulator is used within a module is up to the implementors of the module.
 
-## Exit points
-
-The notion of an exit point is a bit vague, because sending a stanza out of MongooseIM doesn't necessarily mean that processing of the accumulator is terminated.
-A single stanza entering the system creates an acc, but then may result in multiple stanzas going out - for example changing a privacy list triggers presence updates and a roster push.
-It is in fact hard to determine when the process has really been finished.
-The approach is to pass along one acc and, when something is just about to be sent out, either send a text and return the acc or pass on a clone of the acc and return the original one.
-This gives us an option to record a fact that a stanza has been sent out.
-Every sending function calls `mongoose_acc:record_sending`, which gives you an opportunity to later consult the `send_result` key to check what send attempts have been made and with what effect.
-When the last stanza in the whole process is leaving the system we have a complete record of what we have sent in the meantime.
-
-An 'exit point' is where we call a function sending something out of the system and, if all goes well, returning the original accumulator possibly with an additional track record of what's just been done.
-
-See the diagram to examine this concept.
-Note that Acc1 going to the routing has the same `ref` as the one returning to the next function:
-
-![You should see an image here; if you don't, use plantuml to generate it from accum_path.uml](accum_path.png)
-
-
-An 'exit point' function should be called with an acc AND a stanza to be sent out. 
-Sometimes the stanza is the original one (which is stored in the accumulator as `element`), sometimes it's not.
-The 'exit point' should return an accumulator.
-
-### `ejabberd_c2s:send_element/3`
+## IQs and accumulators
 
-Here we send a stanza to an XMPP client.
-It is often called directly by a c2s process when it needs to respond to its "owner".
+`mongoose_iq` module exposes dedicated API for accessing IQ-related accumulator fields. These are:
```

Comment:
`mongoose_iq` module exposes a dedicated API for accessing IQ-related accumulator fields. These are: